ql/src/test/results/clientpositive/llap/drop_histogram_stats_for_columns.q.out

```diff
@@ -0,0 +1,307 @@
+PREHOOK: type: DESCTABLE
+PREHOOK: Input: default@test_stats
+POSTHOOK: query: describe formatted test_stats b
+POSTHOOK: type: DESCTABLE
+POSTHOOK: Input: default@test_stats
+col_name            	b                   
+data_type           	int                 
+min                 	2                   
+max                 	3                   
+num_nulls           	0                   
+distinct_count      	2                   
+avg_col_len         	                    
+max_col_len         	                    
+num_trues           	                    
+num_falses          	                    
+bit_vector          	HL                  
+histogram           	Q1: 2, Q2: 2, Q3: 2 
+comment             	from deserializer   
+COLUMN_STATS_ACCURATE	{\"BASIC_STATS\":\"true\",\"COLUMN_STATS\":{\"a\":\"true\",\"b\":\"true\",\"c\":\"true\"}}
+PREHOOK: query: describe formatted test_stats c
+PREHOOK: type: DESCTABLE
+PREHOOK: Input: default@test_stats
+POSTHOOK: query: describe formatted test_stats c
+POSTHOOK: type: DESCTABLE
+POSTHOOK: Input: default@test_stats
+col_name            	c                   
+data_type           	double              
+min                 	1.1                 
+max                 	6.1                 
+num_nulls           	0                   
+distinct_count      	6                   
+avg_col_len         	                    
+max_col_len         	                    
+num_trues           	                    
+num_falses          	                    
+bit_vector          	HL                  
+histogram           	Q1: 2.1, Q2: 3.1, Q3: 5.1
+comment             	from deserializer   
+COLUMN_STATS_ACCURATE	{\"BASIC_STATS\":\"true\",\"COLUMN_STATS\":{\"a\":\"true\",\"b\":\"true\",\"c\":\"true\"}}
+PREHOOK: query: alter table test_stats drop statistics for columns a, c
+PREHOOK: type: ALTERTABLE_DROP_COL_STATS
+PREHOOK: Input: default@test_stats
+PREHOOK: Output: default@test_stats
+POSTHOOK: query: alter table test_stats drop statistics for columns a, c
+POSTHOOK: type: ALTERTABLE_DROP_COL_STATS
+POSTHOOK: Input: default@test_stats
+POSTHOOK: Output: default@test_stats
+PREHOOK: query: describe formatted test_stats
+PREHOOK: type: DESCTABLE
+PREHOOK: Input: default@test_stats
+POSTHOOK: query: describe formatted test_stats
+POSTHOOK: type: DESCTABLE
+POSTHOOK: Input: default@test_stats
+# col_name            	data_type           	comment             
+a                   	string              	                    
+b                   	int                 	                    
+c                   	double              	                    
+	 	 
+# Detailed Table Information	 	 
+Database:           	default             	 
+#### A masked pattern was here ####
+Retention:          	0                   	 
+#### A masked pattern was here ####
+Table Type:         	MANAGED_TABLE       	 
+Table Parameters:	 	 
+	COLUMN_STATS_ACCURATE	{\"BASIC_STATS\":\"true\",\"COLUMN_STATS\":{\"a\":\"true\",\"b\":\"true\",\"c\":\"true\"}}
```

Comment:
Why are column stats accurate after drop?

Reply:
If I understand correctly, we need to do this in HS2 - after `getMSC().deleteColumnStatistics(request);` returns successfully in `Hive#deleteColumnStatistics()`.

I think we can probably use `StatsSetupConst#removeColumnStatsState(Map<String, String> params, List<String> colNames)` to update the column stats for the list of columns in the ALTER statement. However, for partitioned tables, when partition info is not provided in the ALTER statement, we need to fetch ALL partitions (using `List<Partition> getPartitions(Table tbl)` in Hive.java), and then for each Partition, we need to call `StatsSetupConst#removeColumnStatsState`.

Do you think there is a better and more efficient way to do this?

Reply:
I think my analysis above is not correct. Columns stats are accurate even after dropping stats because the `Table`'s parameters still has the stats marked as accurate. I am trying to figure out how to update the parameters in HMS so that `getTable()` API in `Hive.java` can return `Table` with updated parameters.

Reply:
Yes, maybe this should be handled by HMS. AFAIK stats autogather also update them using alter table HMS api call. 
https://github.com/apache/hive/blob/db041fd7b71a15a8bd1a997b0908a02e8124d7ac/ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java#L306